rdflib/contrib/rdf4j/client.py

```diff
@@ -0,0 +1,1193 @@
+"""RDF4J client module."""
+
+from __future__ import annotations
+
+import contextlib
+import io
+import typing as t
+from dataclasses import dataclass
+from typing import Any, BinaryIO, Iterable
+
+import httpx
+
+from rdflib import BNode
+from rdflib.contrib.rdf4j.exceptions import (
+    RDF4JUnsupportedProtocolError,
+    RDFLibParserError,
+    RepositoryAlreadyExistsError,
+    RepositoryError,
+    RepositoryFormatError,
+    RepositoryNotFoundError,
+    RepositoryNotHealthyError,
+    TransactionClosedError,
+    TransactionCommitError,
+    TransactionPingError,
+    TransactionRollbackError,
+)
+from rdflib.contrib.rdf4j.util import (
+    build_context_param,
+    build_infer_param,
+    build_sparql_query_accept_header,
+    build_spo_param,
+    rdf_payload_to_stream,
+)
+from rdflib.graph import DATASET_DEFAULT_GRAPH_ID, Dataset, Graph
+from rdflib.query import Result
+from rdflib.term import IdentifiedNode, Literal, URIRef
+
+SubjectType = t.Union[IdentifiedNode, None]
+PredicateType = t.Union[URIRef, None]
+ObjectType = t.Union[IdentifiedNode, Literal, None]
+
+
+@dataclass(frozen=True)
+class NamespaceListingResult:
+    """RDF4J namespace and prefix name result."""
+
+    prefix: str
+    namespace: str
+
+
+class NamespaceManager:
```

Comment:
Could you consider an alternate name here to ensure no confusion with RDFLib's NamespaceManager, or are you following naming suggested by the RDF4J API?

Reply:
Done in https://github.com/RDFLib/rdflib/pull/3306/commits/174e9f91455d39bc94cfe654d76eaa037d4db853.